src/gpodder/plugins/itunes.py

```diff
@@ -69,6 +71,30 @@ def __init__(self):
         self.priority = directory.Provider.PRIORITY_SECONDARY_SEARCH
 
     def on_search(self, query):
-        json_url = 'https://itunes.apple.com/search?media=podcast&term={}'.format(urllib.parse.quote(query))
-
-        return [directory.DirectoryEntry(entry['collectionName'], entry['feedUrl'], entry['artworkUrl100']) for entry in util.read_json(json_url)['results']]
+        offset = 0
+
+        while True:
+            json_url = f'https://itunes.apple.com/search?media=podcast&term={urllib.parse.quote(query)}&limit={PAGE_SIZE}&offset={offset}'
+            json_data = util.read_json(json_url)
+
+            if json_data['resultCount'] > 0:
+                for entry in json_data['results']:
+                    if entry.get('feedUrl') is None:
+                        continue
+
+                    title = entry['collectionName']
+                    url = entry['feedUrl']
+                    image = entry['artworkUrl100']
+
+                    yield(directory.DirectoryEntry(title, url, image))
```

Comment:
[`yield`](https://docs.python.org/3/reference/expressions.html#yield-expressions) is a keyword, not a function:

```suggestion
                    yield directory.DirectoryEntry(title, url, image)
```